processor/processor_test.go

```diff
@@ -0,0 +1,1532 @@
+			callDestProcessing := c.mockStatDestProcessing.EXPECT().Start().Times(1).After(callMarshallSingularEventsEnd)
+			callDestANumEvents := c.mockEnabledADestStats.numEvents.(*mocksStats.MockRudderStats).EXPECT().Count(4).Times(1).After(callDestProcessing)
+			c.mockEnabledADestStats.destTransform.(*mocksStats.MockRudderStats).EXPECT().Start().Times(1).After(callDestProcessing)
+			c.mockEnabledADestStats.numOutputEvents.(*mocksStats.MockRudderStats).EXPECT().Count(2).Times(1).After(callDestProcessing)
+			c.mockEnabledADestStats.destTransform.(*mocksStats.MockRudderStats).EXPECT().End().Times(1).After(callDestProcessing)
+			c.mockEnabledBDestStats.numEvents.(*mocksStats.MockRudderStats).EXPECT().Count(2).Times(1).After(callDestProcessing)
+			c.mockEnabledBDestStats.destTransform.(*mocksStats.MockRudderStats).EXPECT().Start().Times(1).After(callDestProcessing)
+			c.mockEnabledBDestStats.numOutputEvents.(*mocksStats.MockRudderStats).EXPECT().Count(2).Times(1).After(callDestProcessing)
+			c.mockEnabledBDestStats.destTransform.(*mocksStats.MockRudderStats).EXPECT().End().Times(1).After(callDestProcessing)
+			callUserTransformStart := c.mockEnabledBDestStats.userTransform.(*mocksStats.MockRudderStats).EXPECT().Start().Times(1).After(callDestProcessing)
+
+			// We expect one call to user transform for destination B
+			callUserTransform := mockTransformer.EXPECT().Transform(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(1).After(callUserTransformStart).
+				DoAndReturn(func(clientEvents []transformer.TransformerEventT, url string, batchSize int, breakIntoBatchWhenUserChanges bool) transformer.ResponseT {
+					Expect(breakIntoBatchWhenUserChanges).To(BeFalse())
+					Expect(url).To(Equal("http://localhost:9090/customTransform"))
+
+					outputEvents := make([]transformer.TransformerResponseT, 0)
+
+					for _, event := range clientEvents {
+						event.Message["user-transform"] = "value"
+						outputEvents = append(outputEvents, transformer.TransformerResponseT{
+							Output: map[string]interface{}{
+								"message": map[string]interface{}(event.Message),
+							},
+						})
+					}
+
+					return transformer.ResponseT{
+						Events:  outputEvents,
+						Success: false, // ??? response success is ignored in user transform
+					}
+				})
+
+			c.mockEnabledBDestStats.userTransform.(*mocksStats.MockRudderStats).EXPECT().End().Times(1).After(callUserTransform)
+
+			transformExpectations := map[string]transformExpectation{
+				DestinationIDEnabledA: {
+					events:          4,
+					messageIds:      "message-1,message-2,message-3,message-4",
+					receiveMetadata: true,
+				},
+				DestinationIDEnabledB: {
+					events:     2,
+					messageIds: "message-3,message-4",
+				},
+			}
+
+			assertDestinationTransform := func(destinationID string, destinationDefinitionName string) func(clientEvents []transformer.TransformerEventT, url string, batchSize int, breakIntoBatchWhenUserChanges bool) transformer.ResponseT {
```

Comment:
This same function definition repeated 3 times in this file. If we can move this function out of this particular test and use it across all tests in this file, it becomes easy to read. But is that possible?